app/assets/javascripts/to_do_items.coffee

```diff
@@ -0,0 +1,72 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
+
+class @TodoItem
+  initialize: () ->
+    bindAddEvents()
+
+  onCreate: (options) ->
+    removeCallout(options.listId)
+    appendNewOne(options.listId, options.html)
+    focusOn(options.listId)
+    highlightNewOne(options.id) if options.id
+
+
+  appendNewOne = (listId, html) ->
+    $('#list-' + listId + ' ul').append(html)
+
+  focusOn = (listId) ->
+    makeFocusOn $('#form-' + listId + " input[type='text']")
+
+  highlightNewOne = (id) ->
+    $('#item-' + id).css({'background-color':'#ffffe0'}).
+      animate({'background-color':'#fff'}, 2000)
+
+  removeCallout = (listId ) ->
+    $('#list-' + listId + ' div.bs-callout').remove()
+
+  bindAddEvents = () ->
+    $.each $('div.list a'), (_, value) =>
+      expandOnClick value
+
+  bindCancelEvent = (form) ->
+    collapseOnClick form.find('button')
+
+  bindEscKeyEvent = (form) ->
+    form.find("input[type='text']").keypress (event) ->
+      form.find('button').click() if event.keyCode == 27
+
+  initializeForm = (form) ->
+    makeFocusOn form.find("input[type='text']")
+    bindCancelEvent(form)
+    bindEscKeyEvent(form)
```

Comment:
@vlasiak, I'm not a fan of passing a parameter down through the methods. Once there is a single way to get/detect the form (for example via method) we could use it where it is needed instead of passing the parameter.